Add steps on migrating external data disk (#32343)
home-assistant/home-assistant.github.io: source/_includes/common-tasks/data_disk.md

Using an USB attached SSD can draw quite some power. For instance on Raspberry Pi 3 the official Raspberry Pi power supply (PSU) only provides 2.5A which can be too tight. Use a more powerful power supply if you experience issues. Alternatively use a powered USB hub. Connect the Hub to one of the USB slots of your Raspberry Pi, and connect the SSD to the Hub. In this setup the power supply of the Hub will power the attached device(s).

</div>

### Migrating an external data disk to another system

This section shows how to move an external data disk from one system to another.
This can be an option if the following elements apply to your use case:

- You already have a functioning Home Assistant instance (system 1) that is using an external data disk.

> - You have another, new, functioning Home Assistant instance (system 2).

Reviewer: "Functioning" sounds like as if it must be configured or something. I'd just drop that, at least for the target system.

```suggestion
- You have another, new, Home Assistant instance (system 2).
```

- You want now want to use the data disk of system 1 on system 2 instead.

The aim is to migrate the data from system 1 to system 2. One way to do this is by [restoring a backup](/common-tasks/os/#restoring-a-backup). The other way is to move the data disk. This can be an interesting option if you have a large amount of data on your external disk or if your external disk has more storage capacity than your new system.

#### Prerequisites

> - 1 Home Assistant instance using an external data disk (system 1)
> - 1 Home Assistant instance to which you want to move the external data disk (system 2)

Reviewer: Personal preference? :sweat_smile: 

```suggestion
- A Home Assistant instance using an external data disk (system 1)
- A Home Assistant instance to which you want to move the external data disk (system 2)
```

#### To migrate an external data disk to another system

To migrate an external data disk from one system to another, follow these steps:

> 1. [Create a backup](/common-tasks/os/#backups) of both systems and store these backups on another system.

Reviewer: We advertise the feature as a quick way to migrate, but then still ask for a backup :sweat_smile: 

It does make sense, but maybe we can weaken it a bit, e.g.

```suggestion
1. [Create a backup](/common-tasks/os/#backups) of both systems and store these backups on another system (not strictly necessary, but recommended just in case, at least for the important data).
```

2. Shut down system 1 and remove the data disk.

> 3. Make sure system 2 is up and running an has the required version of operating system installed.

Reviewer: We don't really specify a required version. But it is probably good to state here that the OS must be generally installed first:

```suggestion
3. Make sure system 2 has Home Assistant OS installed, and Home Assistant is up and running. Home Assistant is using the data disk (partition) on the boot drive (e.g. SD card) at this point.
```

4. Make sure system 2 has completed the basic [onboarding](/getting-started/onboarding/) steps, including the last steps where devices are discovered automatically.
5. Plug the external disk into system 2 and go to the **Settings** > **System**. Select the three dots menu, and **Restart Home Assistant** > **Reboot system**.

>    **Result**: After reboot, a repair issue is displayed **Multiple data disks detected**.

Author: ```suggestion
   **Result**: A repair issue is displayed **Multiple data disks detected**.
```

>    - The repair issue comes up because system 2 now sees two file systems with an identical name. During a reboot, there is a name conflict and it is undefined which file system should be used. This can lead to system instabilities.

Reviewer: ```suggestion
   - The repair issue comes up because system 2 now sees two file systems with an identical name. During a reboot, there is a name conflict with the existing data disk as it is undefined which file system should be used. This can lead to a random selection of the system you end up with. Hence you must make a decision.
```

>    - The repair issue comes up because system 2 now sees two file systems with an identical name. After a reboot, there is a name conflict and it is undefined which file system should be used. This can lead to system instabilities.

Author: ```suggestion
   - The repair issue comes up because system 2 now sees two file systems with an identical name. During a reboot, there is a name conflict and it is undefined which file system should be used. This can lead to system instabilities.
```

>    - The repair issue comes up because system 2 now sees two file systems with an identical name. After a reboot, there is a name conflict and it is undefined which file system should be used. This can lead to system instabilities.

Reviewer: Just realized that the fact that we don't detect this at runtime and need to do a reboot makes the whole operation actually unpredicatable :cold_face: 

E.g. on that startup, the system might pick the internal or external data disk!

I think we have to make this hot-pluggable capable first.

/cc @mdegat01 

```suggestion
   - The repair issue comes up because system 2 now sees two file systems with an identical name. After a reboot, there is a name conflict and it is undefined which file system should be used. This can lead to system instabilities.
```

Reviewer: With https://github.com/home-assistant/supervisor/pull/5023 this is now implemented in Supervisor. Since this is targeting next, we can adjust the documentation and drop the reboot.

</div>

### Migrating an external data disk to another system

This section shows how to move an external data disk from one system to another.
This can be an option if the following elements apply to your use case:

- You already have a functioning Home Assistant instance (system 1) that is using an external data disk.
- You have another, new, functioning Home Assistant instance (system 2).
- You want now want to use the data disk of system 1 on system 2 instead.

The aim is to migrate the data from system 1 to system 2. One way to do this is by [restoring a backup](/common-tasks/os/#restoring-a-backup). The other way is to move the data disk. This can be an interesting option if you have a large amount of data on your external disk or if your external disk has more storage capacity than your new system.

#### Prerequisites

> - Home Assistant Operating System 13.0 or later

Reviewer 2: 13 does not yet exist?
https://github.com/home-assistant/operating-system/releases

Generally we do not note version numbers, as that is done here, should it mention core 2024.5.0+?
